src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -811,14 +802,10 @@ bool ShenandoahControlThread::service_stw_degenerated_cycle(GCCause::Cause cause
   } else {
     assert(_degen_generation->is_young(), "Expected degenerated young cycle, if not global.");
     ShenandoahOldGeneration* old = heap->old_generation();
-    if (old->state() == ShenandoahOldGeneration::BOOTSTRAPPING && !gc.upgraded_to_full()) {
+    if (old->state() == ShenandoahOldGeneration::BOOTSTRAPPING) {
```

Comment:
An assert has been added in `ShenandoahFullGC` to be certain the old generation is put in the `WAITING_FOR_BOOTSTRAP` state after a Full GC>